src/test/java/io/camunda/zeebe/bpmnassert/testengine/db/InMemoryZeebeDbTransactionTest.java

```diff
@@ -0,0 +1,479 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.bpmnassert.testengine.db;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.camunda.zeebe.db.*;
+import io.camunda.zeebe.db.impl.DbLong;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public final class InMemoryZeebeDbTransactionTest {
```

Comment:
🔧 An interesting case (which I believe will fail) would be to start a transaction. In the transaction a key gets deleted and the same key get re-added:
```java
columnFamily.delete(key);
columnFamily.put(key, value);
```
In this case I'd expect the key to be available since we execute the put later than the delete. Yet, I'm fairly sure the key will not exist since on commit we will first process all the puts, and after that all the deletes.

Reply:
Nice one. If that fails, we should also tell Chris and Philipp, cause this may also happen in EZE.